File: src/frontend/src/types/torrent.ts
export interface TorrentResult {
  title: string
  size: number
  seeders: number
  leechers: number
  age: string
  category: string
  source: string
  magnet: string
  infoHash?: string
}

export type PluginStatus = 'idle' | 'loading' | 'healthy' | 'cached' | 'degraded' | 'error'

export interface PluginHealth {
  name: string
  version: string
  status: PluginStatus
  failures: number
  lastError?: string | null
  degradedUntil?: number | null
  rateLimit?: number
  sandboxed?: boolean
  resultCount?: number
}

File: release/backend/_internal/models/torrent.py
from pydantic import BaseModel
from typing import Optional

class TorrentResult(BaseModel):
    title: str
    size: int  # Size in bytes
    seeders: int
    leechers: int
    age: str
    category: str
    source: str
    magnet: str
    infoHash: Optional[str] = None
